src/main/java/org/fogbowcloud/saps/engine/core/archiver/Archiver.java

```diff
@@ -156,13 +154,11 @@ private void deleteTempData(SapsImage task) {
             try {
                 //TODO Remove archive task from here
                 if (this.executionDebugMode && task.getState().equals(ImageTaskState.FAILED)) {
-                    permanentStorage.archive(task);
+                    permanentStorage.archive(task.getTaskId(), task.isFailed());
```

Comment:
update the doc of the permanentstorage interface to explain the meaning of the boolean attribute. I'd also suggest that we should pass only the taskId. The implementation of the PermanentStorage should discover the task is failed or not. If you don't want to refactor that in this PR, add a FIXME into the interface (note that, if we thank it is too much responsibility on the shoulders of the PermanentStorage, we should rollback and pass the Task object to all three methods of the interface, instead of the id)

Reply:
I rolling back the Permanent Storage methods parameters, passing the Task Object. 
Regarding booleans, I believe that we should remove them. In most methods, `True` is always returned.

Reply:
I added a FIXME to remove the boolean return.